README.md

```diff
@@ -85,5 +85,86 @@ iex> EchoClient.echo(pid, "Hi")
 *DBG* #PID<0.379.0> received frame: {:text, "Hi"}
 ```
 
+#### start_link/3 && start/3 
+
+You can start a supervisioned WebSockex connection by using the regular **start_link/3**:
+
+```elixir
+iex> {:ok, pid} = WebSockex.start_link(url, __MODULE__, state)
+```
+
+And an unsupervisioned one (meaning it won't try to re-establish the connection on an exceptional close) with **start/3**:
+
+```elixir
+iex> {:ok, pid} = WebSockex.start(url, __MODULE__, state)
+```
+
+This can be useful if you have don't want the socket to automatically try reconnection, you have no need for the built-in supervision or if you're implementing your own supervision strategy.
+
+
+#### Callbacks
+###### terminate/2
+
+This callback allows you to handle different events related to the WebSocket termination. WebSockex will provide you a 2-element tuple for this event, e.g. `{:remote, :closed}`, and while usually you'll want to negotiate and handle the close event, as per WS Spec, there might be cases where you simply want the socket to exit as if it was a normal event. In those case you can return `exit(:normal)` and no exception will be raised
+
+```elixir
+def terminate(reason, state) do
+    IO.puts(\nSocket Terminating:\n#{inspect reason}\n\n#{inspect state}\n")
+    exit(:normal)
+end
+```
+
+
+#### Example Usage With a Chrome Remote Debugging Enabled Instance (or driver), and using Poison to decode/encode the remote-protocol message passing (assuming you have started chrome with a remote debugging port, and have the WebSocket address for the page/tab in question), closing chrome would cause {:remote, :closed} to be raised, but in this case we really don't care:
+
+```elixir
+defmodule ChromeDebugger do
+     use WebSockex
+     
+     def start(url, state) do
+       {:ok, pid} = WebSockex.start(url, __MODULE__, state)
+       WebSockex.send_frame(pid, {:text, Poison.encode!(%{id: 1, method: "Network.enable", params: %{}})})
+       WebSockex.send_frame(pid, {:text, Poison.encode!(%{id: 2, method: "Runtime.enable", params: %{}})})
+       WebSockex.send_frame(pid, {:text, Poison.encode!(%{id: 3, method: "Page.enable", params: %{}})})
+       {:ok, pid}
+     end
+
+     def terminate(reason, state) do
+       exit(:normal)
+     end
+
+     def handle_frame({_type, msg}, state) do
+       case Poison.decode(msg) do
+         {:error, error} ->
+           Logger.debug(inspect msg)
+           Logger.error(inspect error)
+         {:ok, message} ->
+           case message["method"] do
+             "Network.webSocketFrameReceived" ->
+                with true <- message["params"]["response"]["opcode"] == 1,
+                   message_pl <- message["params"]["response"]["payloadData"]
+                do
+                  IO.inspect(message_pl)
+                else
+                  _ -> :ok
+                end
+              _ -> :ok
+           end
+        end
+        {:ok, state}
+     end
+end
+```
```

Comment:
This is actually pretty neat. You should throw it in an example!

Then put the bold preface in a `@moduledoc` section and link to it in your terminate example.

Reply:
Ok - I was going to write a more complete example, but it requires so much non-related stuff that I decided it didn't make sense.